### PR TITLE
Add support for non sentence transformer models

### PR DESCRIPTION
**Pull request by timovdk**
Add support for (almost) any HF model. This PR also contains one example of this, XLM-RoBERTa-large.

There is quite some reused code between `hf_embeddings.py` and `sentence_embeddings.py`, ideally I'll move some functions into a `utils.py` or some `base.py` file. Open to suggestions :)

### CODE REVIEW

**Comment by timovdk** [comment]
@J535D165, @jteijema What do you think of mocking the SentenceTransformer class, such that fit_transform() returns some cached embeddings for each feature extractor, preventing downloading each feature extractor? This way we can still test our own normalization and quantization implementations, but we don't have to download the full models.
If you have other ideas, please share them haha.

**Comment by J535D165** [comment]
Let's not test the large things. Can't you replace it with a smaller model during unit testing?

Another thing you can do is to make a model matrix in the actions file (I did this in Datahugger). Not all cases need testing against two Python versions and two platforms.
